patches/api/0117-LivingEntity-Active-Item-API.patch

```diff
@@ -0,0 +1,159 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Fri, 29 Jun 2018 00:19:19 -0400
+Subject: [PATCH] LivingEntity Active Item API
+
+API relating to items being actively used by a LivingEntity
+such as a bow or eating food.
+
+Co-authored-by: Jake Potrebic <jake.m.potrebic@gmail.com>
+
+diff --git a/src/main/java/org/bukkit/entity/HumanEntity.java b/src/main/java/org/bukkit/entity/HumanEntity.java
+index 2308fa3ca898bcb6c0ac2d4853f82a3398bf51f3..15115b1049bc5053796b84539acbf576bcba1a5f 100644
+--- a/src/main/java/org/bukkit/entity/HumanEntity.java
++++ b/src/main/java/org/bukkit/entity/HumanEntity.java
+@@ -319,7 +319,9 @@ public interface HumanEntity extends LivingEntity, AnimalTamer, InventoryHolder
+      * blocking).
+      *
+      * @return Whether their hand is raised
++     * @see LivingEntity#hasActiveItem()
+      */
++    @org.jetbrains.annotations.ApiStatus.Obsolete(since = "1.20.4") // Paper - active item API
+     public boolean isHandRaised();
+ 
+     /**
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index 4a4a749449bc561a73e6747386c8ad51e623fc1e..3a9580f9f018be20ca27332427476e7e15321a47 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -202,15 +202,19 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      *
+      * @return the item being used by the player, or null if they are not using
+      * an item
++     * @deprecated Use {@link #getActiveItem()}
+      */
+     @Nullable
++    @Deprecated(forRemoval = true, since = "1.20.4") // Paper
+     public ItemStack getItemInUse();
+ 
+     /**
+      * Gets the number of ticks remaining for the current item's usage.
+      *
+      * @return The number of ticks remaining
++     * @deprecated use {@link #getActiveItemRemainingTime()}
+      */
++    @Deprecated(forRemoval = true, since = "1.20.4") // Paper
+     public int getItemInUseTicks();
+ 
+     /**
+@@ -219,7 +223,9 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      * or throwing a trident.
+      *
+      * @param ticks The number of ticks remaining
++     * @deprecated use {@link #setActiveItemRemainingTime(int)}
+      */
++    @Deprecated(forRemoval = true, since = "1.20.4") // Paper
+     public void setItemInUseTicks(int ticks);
+ 
+     /**
+@@ -850,4 +856,100 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      */
+     void setShieldBlockingDelay(int delay);
+     // Paper end
++
++    // Paper start - active item API
```

Comment:
Letter case are not consistent across all descriptions of return/param in the methods of this block